### PR TITLE
Fix `selector-combinator-allowed-list` reported ranges

### PR DESCRIPTION
**Pull request by ryo-manba**
> Which issue, if any, is this issue related to?

See https://github.com/stylelint/stylelint/issues/7904

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/rules/selector-combinator-allowed-list/index.mjs
Comment (Mouvedia) [inline lib/rules/selector-combinator-allowed-list/index.mjs:1]: refactor
off topic

```js
					const { value } = combinatorNode;
					const normalizedValue = normalizeCombinator(value);
					if (primary.includes(normalizedValue)) {
						return;
					}
					const { sourceIndex: index, raws } = combinatorNode;
```

Could become

```js
					const { raws, sourceIndex: index, value } = combinatorNode;
					const normalizedValue = normalizeCombinator(value);

					if (primary.includes(normalizedValue)) return;
```

**Comment by romainmenke** [comment]
> should `getDeclarationValue` become agnostic?
e.g. accept combinators

I am not sure what you mean? :)
I don't see the connection between declaration values and combinators in selectors.

**Review by romainmenke — APPROVED** [review]
LGTM 🎉

**Comment by Mouvedia** [comment]
> I am not sure what you mean? :)

We could have reused `getFooValue` [here](https://github.com/stylelint/stylelint/pull/8046/files#diff-f310d487cc8a2d8be9f46a961bfb04a6fd4c07441af62d5088d7828815b4df72R51).

**Pull request merged**